### PR TITLE
Move aliases and types to schema overrides

### PR DESCRIPTION
**Pull request by mieciu**
Highlights:
* Moving `TypeMappings` and `Aliases` form `IndexConfiguration` to `SchemaConfiguration` (what used to be called `static-schema`, I'm also changing that to `schemaOverrides`
* Cleanup all the deprecated code logic
* There's more to come with the deprecated field changes, so you might see few line commented out here and there as I didn't want this PR to grow too much and conflict with some @nablaone's recent work

**This might look like only configuration syntax change on the surface**, but it actually aims to reduce the usage of configuration and make the code refer to schema instead.
image

### CODE REVIEW

File: quesma/clickhouse/parser.go [hunk 1]
```diff
@@ -146,6 +146,8 @@ func SchemaToColumns(schemaMapping *schema.Schema, nameFormatter TableColumNameF
 			fType = "Nullable(DateTime64)"
 		case schema.TypeDate.Name:
 			fType = "Nullable(Date)"
+			// TODO if someone sets `type: date` to a field in schemaOverrides AND this is a timestamp field for which we have dedicated logic (use DateTime64 + add DEFAULT now64())
```
Comment (jakozaur) [inline quesma/clickhouse/parser.go:149]: NOT, but TODO are mostly used in form `TODO:`

Comment (jakozaur) [reply quesma/clickhouse/parser.go:149]: Not sure also if I understand `TODO`, they should be in form what is to be fixed and explanation.

File: quesma/clickhouse/table.go [hunk 2]
```diff
@@ -21,8 +21,8 @@ type Table struct {
 	Config       *ChTableConfig
 	Created      bool // do we need to create it during first insert
 	indexes      []IndexStatement
-	aliases      map[string]string //deprecated
-	// we should use aliases directly from configuration, not store them here
+	aliases      map[string]string //deprecated TODO looks like we've reached the point where this becomes a problem
```
Comment (jakozaur) [inline quesma/clickhouse/table.go:24]: E.g. `TODO: Remove before first stable release. Use field with type alias instead.

**Review by jakozaur — APPROVED** [review]
Minor comments, someone else should also reviewed it.

File: examples/kibana-sample-data/quesma/config/local-dev.yaml [hunk 3]
```diff
@@ -31,34 +31,41 @@ processors:
...
-          static-schema:
+          schemaOverrides:
             fields:
+              timestamp:
+                type: alias
+                targetColumnName: "@timestamp"
               message:
-                type: "text"
-              host.name:
-                type: "keyword"
```
Comment (avelanarius) [inline examples/kibana-sample-data/quesma/config/local-dev.yaml:52]: Was there some problem with `host.name` that you decided to delete this?

**Review by avelanarius — APPROVED** [review]

File: examples/kibana-sample-data/quesma/config/local-dev.yaml (hunk 3 above)
Comment (mieciu) [reply examples/kibana-sample-data/quesma/config/local-dev.yaml]: it just slipped unnoticed :) bringing it back

File: examples/kibana-sample-data/quesma/config/local-dev.yaml [hunk 4]
```diff
@@ -31,34 +31,43 @@ processors:
...
+              timestamp:
+                type: alias
+                targetColumnName: "@timestamp"
               message:
-                type: "text"
-              host.name:
+                type: text
+              "host.name":
                 type: "keyword"
               service.name:
```
[the remark this answers is not in the document]
Comment (mieciu) [reply examples/kibana-sample-data/quesma/config/local-dev.yaml]: good catch!

**Review by pdelewski — APPROVED** [review]

**Pull request merged**